inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/view/InAppMessageViewListener.kt

```diff
@@ -73,13 +73,24 @@ internal class InAppMessageViewListener(
             // If user only checked the opt-out box, just assign the isOptOutChecked variable.
             this.isOptOutChecked = (view as CheckBox).isChecked
         } else {
+
+            // Disable closing the message if not dismissable.
+            if (message != null && !message.isCampaignDismissable()) {
+                return
+            }
             // Handling button click in coroutine.
             handleClick(view.id)
         }
     }
 
+    @SuppressWarnings("ReturnCount")
     override fun onKey(v: View?, keyCode: Int, event: KeyEvent?): Boolean {
         if (event != null && event.action == KeyEvent.ACTION_UP && keyCode == KeyEvent.KEYCODE_BACK) {
+
+            // Disable closing the message if not dismissable.
+            if (message != null && !message.isCampaignDismissable()) {
+                return false
```

Comment:
This should return true since it will fall to other `onKey` listener that may possibly close the campaign.
Can add just if dismissible then `handleClick`

Reply:
this is applicable only for the KeyEvent.KEYCODE_BACK when the campaign is displayed. what other cases are you referring to?

Reply:
if a listener returns false in `onKey` function, the other `onKey` function that are set within an activity will be triggered, and may cause the campaign to be dismissed.

ex.
MainActivity also has overridden `onKey`
So if our `onKey` returns false, it means that our listener did not handle the back button behavior, and will trigger the `onKey` of MainActivity.

Reply:
I'll keep it to true, the BU spec is to keep the back button behavior without closing the campaign.